src/apply.rs

```diff
@@ -224,7 +279,7 @@ where
 }
 
 /// Apply a `Diff` to a base image with default fuzzy matching
-pub fn apply(base_image: &str, diff: &Diff<'_, str>) -> Result<String, ApplyError> {
```

Comment:
I think we should make it `ApplyResult<String, ApplyError>` instead of `Result<ApplyResult<String>, ApplyError>` to make it more readable. This also makes sense because we might want to see how many changes we made to make assertions in tests.